### ISSUE TITLE
Supporting multiple geometries

### ISSUE THREAD

**Issue by ghost** [chinedufn/collada-dae-parser#6]
Hi Francis,

I was wondering if you'd want to support multiple geometries at some point ?

I had a quick look at the source and it looked like it could work by repeating the process once per geometry, I'm happy to have a go at this if it's an addition that you'd like.

Thanks,
Alvin

**Comment by chinedufn** [comment]
Yeah I thought about this recently and I don't think there are any good reasons not to support multiple geometries.

Curious about what your thinking is here, but mine was to just output an array of meshes to a JSON file.

---

That would be absolutely fantastic! Want to give it a go and just ping me if you ever get stuck? (Apologies in advance for any under-commented code).

This is exciting!

**Comment by alvinsight** [comment]
Hey Francis,
That's great to hear! 
Our thinking is probably similar, currently we have : 
```js
{
  jointInverseBindPoses: { ... },
  jointNamePositionIndex: { ... },
  keyframes: {...},
  vertexNormalIndices: [...],
  // etc
}
```

And I'm thinking of either : 
```js
{
  'Cube' : {
    keyframes: {...},
    vertexNormalIndices: [...],
    // etc
  },
  'Character' : {
    jointInverseBindPoses: { ... },
    jointNamePositionIndex: { ... },
    keyframes: {...},
    vertexNormalIndices: [...],
    // etc
  }
}
```
Or 
```js
[
  {
    name : 'Cube',
    keyframes: {...},
    vertexNormalIndices: [...],
    // etc
  },
  {
    name : 'Skeleton',
    jointInverseBindPoses: { ... },
    jointNamePositionIndex: { ... },
    keyframes: {...},
    vertexNormalIndices: [...],
    // etc
  }
]
```
depending on what you prefer (JSON objects or JSON arrays).

I'd be against any sort of arbitrary grouping (like, separating skinned meshes and others) I think of the greatest things about this module is that it's minimal and un-opinionated, and I'd want to keep it this way.

Let me know what your thoughts are,
Thanks,
Alvin

**Comment by chinedufn** [comment]
I 100% agree with you on not wanting to group things. Yup - let's keep things as simple as possible. I'm on board!

In terms of the structure, I'm cool with array or object, whichever you prefer. I'm sure if it ends up feeling weird one way or the other it's an easy change.

---

But otherwise I have no other initial thoughts. I think it just comes down to having a very simple test COLLADA file with 2 geometries, setting up the expected JSON result and then making the test pass.

And if there are any common gotchas / issues while exporting would be great to alert the user. For 

example error- https://github.com/chinedufn/collada-dae-parser/blob/master/src/library_geometries/parse-library-geometries.js#L6-L10

example test for error - https://github.com/chinedufn/collada-dae-parser/blob/master/test/multiple-meshes/blender/multiple-mesh_blender_test.js#L12-L16

But things like that will just come up whenever we run into problems so likely no need to plan for them ahead of time!
